api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java

```diff
@@ -1028,209 +1131,929 @@ public CohortDefinition getIPTB1(
-            + "         WHERE"
-            + "             p.voided = 0 AND e.voided = 0"
-            + "                 AND o.voided = 0"
-            + "                 AND e.location_id = :location"
-            + "                AND e.encounter_type = ${53}   "
-            + "                AND o.concept_id = ${6129}  "
-            + "                AND o.value_datetime IS NOT NULL"
-            + "                AND o.value_datetime BETWEEN DATE_ADD(CAST((SELECT "
-            + "                             ee.encounter_datetime"
-            + "                         FROM"
-            + "                             encounter ee"
-            + "                                 INNER JOIN"
-            + "                             obs oo ON ee.encounter_id = oo.encounter_id"
-            + "                         WHERE"
-            + "                             ee.voided = 0 AND oo.voided = 0"
-            + "                                 AND p.patient_id = ee.patient_id"
-            + "                                 AND ee.encounter_type = ${60}"
-            + "                                 AND oo.concept_id = ${23985}"
-            + "                                 AND oo.value_coded IN (${656} , ${23982})"
-            + "                                 AND ee.location_id = :location"
-            + "                                 AND ee.encounter_datetime <= :endDate"
-            + "                         LIMIT 1)"
-            + "                     AS DATE),"
-            + "                 INTERVAL 173 DAY) AND DATE_ADD(CAST((SELECT "
-            + "                             ee.encounter_datetime"
-            + "                         FROM"
-            + "                             encounter ee"
-            + "                                 INNER JOIN"
-            + "                                obs oo ON ee.encounter_id = oo.encounter_id"
-            + "                            WHERE"
-            + "                                ee.voided = 0 AND oo.voided = 0"
-            + "                                    AND p.patient_id = ee.patient_id"
-            + "                                    AND ee.encounter_type = ${60}"
-            + "                                    AND oo.concept_id = ${23985}"
-            + "                                    AND oo.value_coded IN (${656} , ${23982})"
-            + "                                    AND ee.location_id = :location"
-            + "                                    AND ee.encounter_datetime <= :endDate"
-            + "                            LIMIT 1)"
-            + "                        AS DATE),"
-            + "                    INTERVAL 365 DAY)"
-            + "         GROUP BY p.patient_id";
+        "SELECT "
+            + "                            p.patient_id "
+            + "                        FROM "
+            + "                            patient p "
```

Comment:
Please let's improve this query, ping me when ready

Reply:
This query was reimplemented on TPT Eligible ticket.